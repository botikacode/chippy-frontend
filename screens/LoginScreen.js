import React, {useEffect, useState} from 'react'
import { TouchableOpacity, StyleSheet, View, Text, TextInput} from 'react-native'

import { emailValidator } from '../helpers/emailValidator'
import { passwordValidator } from '../helpers/passwordValidator'
import {getCustomers} from '../db/customersApi'

import Layout from '../components/Layout'
import Logo from '../components/Logo'


export default function LoginScreen({ navigation }) {

  
  const [customers, setCustomers] = useState([])

  const loadCustomers = async () =>{

    const data = await getCustomers()
    setCustomers(data)
  }

  useEffect(() =>{
    loadCustomers()
  }, [])

  
  const [email, setEmail] = useState({ value: '', error: '' })
  const [password, setPassword] = useState({ value: '', error: '' })
  

  const onLoginPressed = () => {
    const emailError = emailValidator(email.value)
    const passwordError = passwordValidator(password.value)
    if (emailError || passwordError) {
      console.log(emailError)
      console.log(passwordError)
      
      setEmail({ ...email, error: emailError })
      setPassword({ ...password, error: passwordError })
      return
    }
    
    customers.map((custom) => {
      if(custom.email === email.value && custom.password === password.value){
        navigation.navigate('TabNavigator', {
          idSesion: 1,
        })
      }else{
        setEmail({ ...email, error: "" })
        setPassword({ ...password, error: "Usuario incorrecto" })
        return
      }
      
    })

    
  }
  return (
    <Layout>
      <Logo />
      
      <Text style={styles.inicioSesion} >Inicio de sesi칩n</Text>
      
      <View style={styles.container}>
      <TextInput
        style={styles.input}
        label="email"
        returnKeyType="next"
        underlineColor="transparent"
        mode="outlined"
        placeholder="Email"
        value={email.value}
        onChangeText={(text) => setEmail({ value: text, error: '' })}
        error={!!email.error}
        errorText={email.error}
        autoCapitalize="nombre"
        autoCompleteType="email"
        textContentType="emailAddress"
        keyboardType="email-address"

        placeholderTextColor="#576574"
        
        
      />
      
      {email.error ? <Text style={styles.error}>{email.error }</Text> : null}
     
      <TextInput
        style={styles.input}
        underlineColor="transparent"
        mode="outlined"
        placeholder="Contrase침a"
        placeholderTextColor="#576574"
        returnKeyType="done"
        value={password.value}
        onChangeText={(text) => setPassword({ value: text, error: '' })}
        error={!!password.error}
        errorText={password.error}
        secureTextEntry
      /> 
      {password.error ? <Text style={styles.error}>{password.error }</Text> : null}
      </View>

      <View style={styles.forgotPassword}>
      <TouchableOpacity>
          <Text style={styles.forgot}>Olvidaste la contrase침a?</Text>
        </TouchableOpacity>
      </View>

      <TouchableOpacity style={styles.buttonCeleste} onPress={() => navigation.navigate("TabNavigator")} /* onPress={onLoginPressed} */>
          <Text style={styles.buttonText}>Iniciar Sesi칩n</Text>
      </TouchableOpacity>

      <View style={styles.row}>
      <Text style={{ color:"#0094FF"}}>No tienes una cuenta? </Text>
      <TouchableOpacity onPress={() => navigation.navigate("RegisterScreen")}>
          <Text style={styles.link}>Registrarme</Text>
        </TouchableOpacity>
      </View>

      {/* reTextEntry
      />
      
      
      <View style={styles.row}>
        <Text>No tienes una cuenta? </Text>
        <TouchableOpacity onPress={() => navigation.replace('RegisterScreen')}>
          <Text style={styles.link}>Registrarme</Text>
        </TouchableOpacity>
      </View> */}
    </Layout>
  )
}

const styles = StyleSheet.create({
  input: {
    width: "70%",
    marginBottom: 7,
    fontSize: 14,
    borderWidth: 1,
    borderColor: "#ced4da",
    height: 30,
    color: "#000000",
    padding: 4,
    borderRadius: 5,
  },
  forgotPassword: {
    width: '65%',
    alignItems: 'flex-end',
    marginBottom: 24,
  },
  forgot: {
    color:"#1234DF",
    fontSize: 11,
  },
  buttonCeleste: {
    paddingTop: 10,
    paddingBottom: 10,
    borderRadius: 5,
    marginBottom: 3,
    backgroundColor: "#0094FF",
    width: "50%",
  },
  buttonText: {
    color: "#fff",
    textAlign: "center",
  },
  link: {
    fontWeight: 'bold',
    color: "#0094FF",
  },
  row: {
    flexDirection: 'row',
    marginTop: 4,
  },
  inicioSesion:{
    color: "#0094FF",
    marginBottom: 10,
    fontSize: 20,
  },
  container: {
    width: '100%',
    marginVertical: 5,
    alignItems: 'center',
  },
  error: {
    fontSize: 10,
    color: '#f13a59'
  },

})
