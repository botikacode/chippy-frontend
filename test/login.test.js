import React from 'react';
import renderer from 'react-test-renderer';
//import {render, screen} from '@testing-library/react'
//import userEvent from '@testing-library/user-event'
//import '@testing-library/jest-dom'
import LoginScreen from '../screens/LoginScreen';

test('renders correctly', () => {
  const tree = renderer.create(<LoginScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

/*test('finds text input', async () => {
  
    // ACT
    await userEvent.click(screen.getByText('Ya tengo una cuenta'))
    await screen.findByRole('heading')
  
    // ASSERT
    expect(screen.getByRole('heading')).toHaveTextContent('hello there')
    expect(screen.getByRole('button')).toBeDisabled()
})*/