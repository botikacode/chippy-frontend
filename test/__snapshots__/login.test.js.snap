// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderiza Login Screen correctamente 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/logo.png",
      }
    }
    style={
      {
        "height": 110,
        "marginBottom": 10,
        "minWidth": 200,
        "width": 110,
      }
    }
  />
  <Text
    style={
      {
        "color": "#0094FF",
        "fontSize": 20,
        "marginBottom": 10,
      }
    }
  >
    Inicio de sesión
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "marginVertical": 5,
        "width": "100%",
      }
    }
  >
    <TextInput
      autoCapitalize="nombre"
      autoCompleteType="email"
      error={false}
      errorText=""
      keyboardType="email-address"
      label="email"
      mode="outlined"
      name="email"
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#576574"
      returnKeyType="next"
      style={
        {
          "borderColor": "#ced4da",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#000000",
          "fontSize": 14,
          "height": 30,
          "marginBottom": 7,
          "padding": 4,
          "width": "70%",
        }
      }
      textContentType="emailAddress"
      underlineColor="transparent"
      value=""
    />
    <TextInput
      error={false}
      errorText=""
      mode="outlined"
      onChangeText={[Function]}
      placeholder="Contraseña"
      placeholderTextColor="#576574"
      returnKeyType="done"
      secureTextEntry={true}
      style={
        {
          "borderColor": "#ced4da",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#000000",
          "fontSize": 14,
          "height": 30,
          "marginBottom": 7,
          "padding": 4,
          "width": "70%",
        }
      }
      underlineColor="transparent"
      value=""
    />
  </View>
  <View
    style={
      {
        "alignItems": "flex-end",
        "marginBottom": 24,
        "width": "65%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#1234DF",
            "fontSize": 11,
          }
        }
      >
        Olvidaste la contraseña?
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#0094FF",
        "borderRadius": 5,
        "marginBottom": 3,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingTop": 10,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "textAlign": "center",
        }
      }
    >
      Iniciar Sesión
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 4,
      }
    }
  >
    <Text
      style={
        {
          "color": "#0094FF",
        }
      }
    >
      No tienes una cuenta? 
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#0094FF",
            "fontWeight": "bold",
          }
        }
      >
        Registrarme
      </Text>
    </View>
  </View>
</View>
`;

exports[`Renderiza Start Screen correctamente 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/logo.png",
      }
    }
    style={
      {
        "height": 110,
        "marginBottom": 10,
        "minWidth": 200,
        "width": 110,
      }
    }
  />
  <Text
    style={
      {
        "color": "#0094FF",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
  >
    ¡Bienvenido a Chippy!
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#0094FF",
        "borderRadius": 5,
        "marginBottom": 3,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingTop": 10,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "textAlign": "center",
        }
      }
    >
      Ya tengo una cuenta
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "marginBottom": 3,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingTop": 10,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#0094FF",
          "textAlign": "center",
        }
      }
    >
      Registrarme
    </Text>
  </View>
</View>
`;

exports[`renders Login Screen correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/logo.png",
      }
    }
    style={
      {
        "height": 110,
        "marginBottom": 10,
        "minWidth": 200,
        "width": 110,
      }
    }
  />
  <Text
    style={
      {
        "color": "#0094FF",
        "fontSize": 20,
        "marginBottom": 10,
      }
    }
  >
    Inicio de sesión
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "marginVertical": 5,
        "width": "100%",
      }
    }
  >
    <TextInput
      autoCapitalize="nombre"
      autoCompleteType="email"
      error={false}
      errorText=""
      keyboardType="email-address"
      label="email"
      mode="outlined"
      name="email"
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#576574"
      returnKeyType="next"
      style={
        {
          "borderColor": "#ced4da",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#000000",
          "fontSize": 14,
          "height": 30,
          "marginBottom": 7,
          "padding": 4,
          "width": "70%",
        }
      }
      textContentType="emailAddress"
      underlineColor="transparent"
      value=""
    />
    <TextInput
      error={false}
      errorText=""
      mode="outlined"
      onChangeText={[Function]}
      placeholder="Contraseña"
      placeholderTextColor="#576574"
      returnKeyType="done"
      secureTextEntry={true}
      style={
        {
          "borderColor": "#ced4da",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#000000",
          "fontSize": 14,
          "height": 30,
          "marginBottom": 7,
          "padding": 4,
          "width": "70%",
        }
      }
      underlineColor="transparent"
      value=""
    />
  </View>
  <View
    style={
      {
        "alignItems": "flex-end",
        "marginBottom": 24,
        "width": "65%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#1234DF",
            "fontSize": 11,
          }
        }
      >
        Olvidaste la contraseña?
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#0094FF",
        "borderRadius": 5,
        "marginBottom": 3,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingTop": 10,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "textAlign": "center",
        }
      }
    >
      Iniciar Sesión
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 4,
      }
    }
  >
    <Text
      style={
        {
          "color": "#0094FF",
        }
      }
    >
      No tienes una cuenta? 
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#0094FF",
            "fontWeight": "bold",
          }
        }
      >
        Registrarme
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders Start Screen correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/logo.png",
      }
    }
    style={
      {
        "height": 110,
        "marginBottom": 10,
        "minWidth": 200,
        "width": 110,
      }
    }
  />
  <Text
    style={
      {
        "color": "#0094FF",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
  >
    ¡Bienvenido a Chippy!
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#0094FF",
        "borderRadius": 5,
        "marginBottom": 3,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingTop": 10,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "textAlign": "center",
        }
      }
    >
      Ya tengo una cuenta
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "marginBottom": 3,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingTop": 10,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#0094FF",
          "textAlign": "center",
        }
      }
    >
      Registrarme
    </Text>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/logo.png",
      }
    }
    style={
      {
        "height": 110,
        "marginBottom": 10,
        "minWidth": 200,
        "width": 110,
      }
    }
  />
  <Text
    style={
      {
        "color": "#0094FF",
        "fontSize": 20,
        "marginBottom": 10,
      }
    }
  >
    Inicio de sesión
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "marginVertical": 5,
        "width": "100%",
      }
    }
  >
    <TextInput
      autoCapitalize="nombre"
      autoCompleteType="email"
      error={false}
      errorText=""
      keyboardType="email-address"
      label="email"
      mode="outlined"
      name="email"
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#576574"
      returnKeyType="next"
      style={
        {
          "borderColor": "#ced4da",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#000000",
          "fontSize": 14,
          "height": 30,
          "marginBottom": 7,
          "padding": 4,
          "width": "70%",
        }
      }
      textContentType="emailAddress"
      underlineColor="transparent"
      value=""
    />
    <TextInput
      error={false}
      errorText=""
      mode="outlined"
      onChangeText={[Function]}
      placeholder="Contraseña"
      placeholderTextColor="#576574"
      returnKeyType="done"
      secureTextEntry={true}
      style={
        {
          "borderColor": "#ced4da",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#000000",
          "fontSize": 14,
          "height": 30,
          "marginBottom": 7,
          "padding": 4,
          "width": "70%",
        }
      }
      underlineColor="transparent"
      value=""
    />
  </View>
  <View
    style={
      {
        "alignItems": "flex-end",
        "marginBottom": 24,
        "width": "65%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#1234DF",
            "fontSize": 11,
          }
        }
      >
        Olvidaste la contraseña?
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#0094FF",
        "borderRadius": 5,
        "marginBottom": 3,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingTop": 10,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "textAlign": "center",
        }
      }
    >
      Iniciar Sesión
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 4,
      }
    }
  >
    <Text
      style={
        {
          "color": "#0094FF",
        }
      }
    >
      No tienes una cuenta? 
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#0094FF",
            "fontWeight": "bold",
          }
        }
      >
        Registrarme
      </Text>
    </View>
  </View>
</View>
`;
